Userland/Utilities/strings.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+#include <unistd.h>
+
+#define DEFAULT_MINIMUM_STRING_LENGTH 4
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    TRY(Core::System::pledge("stdio rpath"));
+
+    Vector<StringView> paths;
+    size_t minimum_string_length = DEFAULT_MINIMUM_STRING_LENGTH;
+
+    enum StringOffsetFormat {
+        None = 0,
+        Decimal,
+        Octal,
+        Hexadecimal
+    };
+
+    size_t current_fd_position;
+    size_t current_path_position = 0;
+    bool show_paths = false;
+    StringView string_offset_format_sv;
+    size_t string_offset_format = StringOffsetFormat::None;
+    size_t string_offset_position;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_option(minimum_string_length, "Specify the minimum string length.", nullptr, 'n', "number");
+    args_parser.add_option(show_paths, "Display the path for each matched file.", nullptr, 'p');
+    args_parser.add_option(string_offset_format_sv, "Write offset relative to start of each file in (d)ec, (o)ct, or (h)ex format.", nullptr, 't', "format");
+    args_parser.set_general_help("Write the sequences of printable characters in files or pipes to stdout.");
+    args_parser.add_positional_argument(paths, "File path", "path", Core::ArgsParser::Required::No);
+    args_parser.parse(arguments);
+
+    if (minimum_string_length < 1) {
+        warnln("Invalid minimum string length {}", minimum_string_length);
+        return 1;
+    }
+
+    if (string_offset_format_sv == "d") {
+        string_offset_format = StringOffsetFormat::Decimal;
+    } else if (string_offset_format_sv == "o") {
+        string_offset_format = StringOffsetFormat::Octal;
+    } else if (string_offset_format_sv == "h") {
+        string_offset_format = StringOffsetFormat::Hexadecimal;
+    } else {
+        if (!string_offset_format_sv.is_empty()) {
+            warnln("Invalid format {}", string_offset_format_sv);
+            return 1;
+        }
+    }
+
+    bool all_ok = true;
+
+    Vector<int> fds;
+    if (paths.is_empty()) {
+        TRY(fds.try_append(STDIN_FILENO));
+    } else {
+        for (auto const& path : paths) {
+            int fd;
+            if (path == "-") {
+                fd = 0;
+            } else {
+                auto fd_or_error = Core::System::open(path, O_RDONLY);
+                if (fd_or_error.is_error()) {
+                    all_ok = false;
+                    warnln("Failed to open {}: {}", path, fd_or_error.error());
+                    continue;
+                }
+                fd = fd_or_error.release_value();
+            }
```

Comment:
`Core::Stream::File` already has a function that was made just for this, it would just be a matter of porting the utility to it. :^)

That said, given that we want to write less C code and more idiomatic C++, using that API wouldn't be that bad either way.

Reply:
Ok, that makes sense. The utility now uses the `Core::Stream::File` API. :^)